### ISSUE TITLE
ManifestParser does not load package.manifests from <project>/App_Plugins in some situations

### ISSUE THREAD

**Issue by PerplexDaniel** [umbraco/Umbraco-CMS#13565]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

11.0.0

### Bug summary

In some situations, [ManifestParser](https://github.com/umbraco/Umbraco-CMS/blob/v11/contrib/src/Umbraco.Infrastructure/Manifest/ManifestParser.cs) does not load package.manifest files in `<project>/App_Plugins`.
Note this is **not** `<project>/wwwroot/App_Plugins`, that one seems to work fine in all scenarios.
Umbraco also reads package.manifest files from `<project>/App_Plugins` (outside of `wwwroot`) through the [`UmbracoPluginPhysicalFileProvider`](https://github.com/umbraco/Umbraco-CMS/blob/v11/contrib/src/Umbraco.Web.Common/Plugins/UmbracoPluginPhysicalFileProvider.cs), which is enabled in [`UmbracoApplicationBuilder`](https://github.com/umbraco/Umbraco-CMS/blob/release-11.0.0/src/Umbraco.Web.Common/ApplicationBuilder/UmbracoApplicationBuilder.cs#L101) using the extension method [`IApplicationBuilder.UseUmbracoPluginsStaticFiles()`](https://github.com/umbraco/Umbraco-CMS/blob/d713b383c97ced18ecc53651952484735a011c29/src/Umbraco.Web.Common/Extensions/ApplicationBuilderExtensions.cs#L115).

However, it is possible to hit [`ManifestParser.GetManifests()`](https://github.com/umbraco/Umbraco-CMS/blob/release-11.0.0/src/Umbraco.Infrastructure/Manifest/ManifestParser.cs#L123) *before* the `UmbracoPluginPhysicalFileProvider` is enabled, in which case package.manifest files residing in <project>/App_Plugins are completely ignored and thus any property editors / JavaScript files / CSS files defined in the package.manifests will not be loaded in the backoffice at all. 

See **Steps to reproduce** on how to reproduce this issue. 

### Steps to reproduce

The bug can be reproduced in a new empty Umbraco 11 installation.

We use `IDataTypeService.GetAll()` to trigger the bug but it is likely there are other ways to trigger it, e.g. reading [`ManifestParser.CombinedManifest`](https://github.com/umbraco/Umbraco-CMS/blob/release-11.0.0/src/Umbraco.Infrastructure/Manifest/ManifestParser.cs#L113) directly.

- `dotnet new install Umbraco.Templates::11.0.0`
    - Ensures v11 of the Umbraco Template is installed.
- `dotnet new umbraco -n UmbracoBug`
- `cd UmbracoBug`
- `mkdir App_Plugins\UmbracoBug&&echo {} > App_Plugins\UmbracoBug\package.manifest`
    - This creates an empty package.manifest file in the folder `/App_Plugins/UmbracoBug` (note: **not** in `wwwroot/App_Plugins` but in the project root).
- `dotnet run`
- Install Umbraco with some username/email/password, leaving everything else default
- Create a file `UmbracoBug.cs` in the root of the project with the following contents:
<details>
<summary><strong>UmbracoBug.cs</strong></summary>

```cs

using Umbraco.Cms.Core.Composing;
using Umbraco.Cms.Core.Manifest;
using Umbraco.Cms.Core.Services;

namespace UmbracoBug;

public class MF : IManifestFilter
{
    public void Filter(List<PackageManifest> manifests)
    {
        if (manifests.Count == 0)
            throw new Exception("No Manifests Found");
        else
            Console.WriteLine($"Manifest count: {manifests.Count}");
    }
}

public class MFC : IComposer
{
    public void Compose(IUmbracoBuilder builder)
        => builder.ManifestFilters().Append<MF>();
}

public class C : IComponent
{
    private readonly IDataTypeService _ds;
    public C(IDataTypeService ds) => _ds = ds;
    public void Initialize() => _ds.GetAll().ToArray();
    public void Terminate() {}
}

public class CC : ComponentComposer<C> {}

```

</details>

- `dotnet run`
- Observe the program crashes with the exception `No Manifests Found`.
    - This obviously should not happen since a package.manifest is present at `App_Plugins\UmbracoBug\package.manifest`.
- Comment or remove the line `public class CC : ComponentComposer<C> {}`
- `dotnet run`
- Observe the program boots normally, it will print `Manifest count: 1` in the console when going to `/umbraco`
    - You need to go to /umbraco since manifests are read lazily when needed.

### Expected result / actual result

Loading package.manifest files from <project>/App_Plugins should always work, and not randomly fail for example when any code calls `IDataTypeService.GetAll()` or similar during startup.

**Comment by kjac** [comment]
Hi there @PerplexDaniel - thank you for your very thorough issue report 💪 

I can reproduce this on V11. Strangely I can't reproduce on V10 🤔 

We'll look into this and possibly add it to an upcoming sprint. It certainly needs to be investigated further.

**Comment by PerplexDaniel** [comment]
@kjac Yeah it was introduced in v11. In v10 `ManifestParser` did not use `IFileProvider` and just looked on disk directly I believe. Either way the `GetManifests()` method was rewritten in v11.

**Comment by kjac** [comment]
Hi @PerplexDaniel,

Again thank you for bringing this to our attention ❤️ this issue has now been resolved in #13597. The fix is slated for V11.1 👍
